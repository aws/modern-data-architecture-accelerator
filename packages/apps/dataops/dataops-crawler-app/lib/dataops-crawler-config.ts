/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps } from '@aws-caef/app';
import { CrawlerDefinition } from '@aws-caef/dataops-crawler-l3-construct';
import { CaefDataOpsConfigParser, CaefDataOpsConfigContents } from '@aws-caef/dataops-shared';
import { Schema } from "ajv";
import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';


export interface GlueCrawlerConfigContents extends CaefDataOpsConfigContents {
    /**
     * Name of the Data Ops project. The crawler config will be autowired to use existing resources deployed by the project.
     */
    projectName: string
    /**
     * Map of crawler names to crawler definitions 
     */
    crawlers: { [ key: string ]: CrawlerDefinition }
}

export class GlueCrawlerConfigParser extends CaefDataOpsConfigParser<GlueCrawlerConfigContents> {
    public readonly crawlerConfigs: { [ key: string ]: CrawlerDefinition }

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )

        this.crawlerConfigs = this.configContents.crawlers

    }

}