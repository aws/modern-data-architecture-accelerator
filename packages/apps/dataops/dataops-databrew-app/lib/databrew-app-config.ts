/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps } from "@aws-caef/app";
import { Schema } from "ajv";
import { Stack } from "aws-cdk-lib";
import * as configSchema from './config-schema.json';
import { DataBrewJobProps, DatasetProps, RecipeProps } from '@aws-caef/dataops-databrew-l3-construct';
import { CaefDataOpsConfigParser, CaefDataOpsConfigContents } from '@aws-caef/dataops-shared';

export interface DataBrewConfigContents extends CaefDataOpsConfigContents {


    // Name of the Data-Ops project.
    readonly projectName: string

    // List of recipes to be created.
    readonly recipes?: {
        [ key: string ]: RecipeProps
    }

    // List of recipes to be created.
    readonly datasets?: {
        [ key: string ]: DatasetProps
    }

    // List of recipes to be created.
    readonly jobs?: {
        [ key: string ]: DataBrewJobProps
    }
}

export class DataBrewConfigParser extends CaefDataOpsConfigParser<DataBrewConfigContents> {

    public readonly jobs?: Record<string, DataBrewJobProps>
    public readonly recipes?: Record<string, RecipeProps>
    public readonly datasets?: Record<string, DatasetProps>

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.jobs = this.configContents.jobs
        this.recipes = this.configContents.recipes
        this.datasets = this.configContents.datasets
    }
}

