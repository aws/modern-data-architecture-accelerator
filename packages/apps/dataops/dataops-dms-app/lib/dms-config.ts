/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import {  CaefAppConfigParserProps } from '@aws-caef/app';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';
import { CaefDataOpsConfigContents, CaefDataOpsConfigParser } from '@aws-caef/dataops-shared';
import { DMSProps } from '@aws-caef/dataops-dms-l3-construct';


export interface DMSConfigContents extends CaefDataOpsConfigContents {
    readonly dms: DMSProps
}


export class DMSConfigParser extends CaefDataOpsConfigParser<DMSConfigContents> {
    public dms: DMSProps
    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.dms = this.configContents.dms
    }
}

