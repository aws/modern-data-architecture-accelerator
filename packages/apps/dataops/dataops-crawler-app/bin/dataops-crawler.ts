#!/usr/bin/env node
/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { GlueCrawlerCDKApp } from "../lib/dataops-crawler";
new GlueCrawlerCDKApp().generateStack()