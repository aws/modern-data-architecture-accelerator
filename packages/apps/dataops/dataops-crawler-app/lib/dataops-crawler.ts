/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { GlueCrawlerL3Construct, GlueCrawlerL3ConstructProps } from '@aws-mdaa/dataops-crawler-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { GlueCrawlerConfigParser } from './dataops-crawler-config';

export class GlueCrawlerCDKApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }
  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const appConfig = new GlueCrawlerConfigParser(stack, parserProps);
    const constructProps: GlueCrawlerL3ConstructProps = {
      ...{
        crawlerConfigs: appConfig.crawlerConfigs,
        securityConfigurationName: appConfig.securityConfigurationName,
        projectName: appConfig.projectName,
        notificationTopicArn: appConfig.projectTopicArn,
      },
      ...l3ConstructProps,
    };
    new GlueCrawlerL3Construct(stack, 'construct', constructProps);
    return [stack];
  }
}
