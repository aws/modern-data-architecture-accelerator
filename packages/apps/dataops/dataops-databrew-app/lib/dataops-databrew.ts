/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParserProps, CaefCdkApp } from '@aws-caef/app';
import { AppProps, Stack } from 'aws-cdk-lib';
import { DataBrewConfigParser } from './databrew-app-config';
import { CaefL3ConstructProps } from '@aws-caef/l3-construct';
import { DataBrewL3Construct, DataBrewL3ConstructProps } from '@aws-caef/dataops-databrew-l3-construct'

export class DataBrewApp extends CaefCdkApp {
    constructor( props: AppProps = {} ) {
        super( "dataops-databrew", props )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: CaefL3ConstructProps, parserProps: CaefAppConfigParserProps ) {
        const appConfig = new DataBrewConfigParser( stack, parserProps )
        const constructProps: DataBrewL3ConstructProps = {
            ...{
                datasets: appConfig.datasets,
                recipes: appConfig.recipes,
                projectName: appConfig.projectName,
                jobs: appConfig.jobs
            }, ...l3ConstructProps
        }
        new DataBrewL3Construct( stack, "jobs", constructProps );
        return [ stack ]
    }
}