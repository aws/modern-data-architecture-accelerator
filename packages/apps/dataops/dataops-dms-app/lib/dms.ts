/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { DMSL3Construct, DMSL3ConstructProps } from '@aws-mdaa/dataops-dms-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { DMSConfigParser } from './dms-config';

export class DMSCDKApp extends MdaaCdkApp {
    constructor( props: AppProps = {} ) {
        super( props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`) )
    }

    protected subGenerateResources ( stack: Stack, l3ConstructProps: MdaaL3ConstructProps, parserProps: MdaaAppConfigParserProps ) {
        const appConfig = new DMSConfigParser( stack, parserProps )

        const constructProps: DMSL3ConstructProps = {
            ...appConfig,
            ...l3ConstructProps
        }
        new DMSL3Construct( stack, "dms", constructProps );
        return [ stack ]
    }
}