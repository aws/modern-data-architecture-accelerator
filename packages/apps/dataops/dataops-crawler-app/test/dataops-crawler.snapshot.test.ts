/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { describe } from '@jest/globals';
import { snapShotTest, snapShotTestApp, Create } from '@aws-mdaa/testing';
import { GlueCrawlerCDKApp } from '../lib/dataops-crawler';
import * as path from 'path';

describe('dataops-crawler Snapshot Tests', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer({
      test: (val: unknown) => typeof val === 'string' && val.includes('[CONFIG:') && val.includes('test-config.yaml]'),
      print: (val: unknown) => {
        const stringVal = val as string;
        return `"${stringVal.replace(/\[CONFIG:[^[\]]*test-config\.yaml\]/, '[CONFIG:test-config.yaml]')}"`;
      },
    });
  });
  snapShotTest(
    'Glue Crawler Stack',
    Create.stackProvider(
      'GlueCrawlerStackMain',
      (_, context) => {
        const moduleApp = new GlueCrawlerCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        return moduleApp.generateStack();
      },
      {
        module_name: 'test-glue-crawler-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );

  snapShotTestApp(
    'Glue Crawler App',
    Create.appProvider(
      context => {
        const moduleApp = new GlueCrawlerCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        moduleApp.generateStack();
        return moduleApp;
      },
      {
        module_name: 'test-glue-crawler-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );
});
