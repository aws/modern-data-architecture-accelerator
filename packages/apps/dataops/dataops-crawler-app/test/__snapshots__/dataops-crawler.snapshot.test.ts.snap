// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataops-crawler Snapshot Tests Glue Crawler App App Snapshot Test 1`] = `
{
  "test-org-test-env-test-domain-test-glue-crawler-main": {
    "Description": "(SO0320-dataops-crawler) Modern Data Architecture Accelerator. Version 1.0.0",
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestclassifiersclassifierCsvC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/classifiers/classifierCsv",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestdatabaseNameexampledatabaseC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/databaseName/example-database",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestdeploymentRoledefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/deploymentRole/default",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestkmsArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/kmsArn/default",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectBucketdefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/projectBucket/default",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectTopicArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/projectTopicArn/default",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaindataopsprojecttestsecurityConfigurationdefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/dataops-project-test/securityConfiguration/default",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
    },
    "Resources": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Stack description parameter to update on version changes",
          "Name": "/test-org/test-domain/test-glue-crawler-main/aws-solution",
          "Tags": {
            "mdaa_cdk_app": "dataops-crawler",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-glue-crawler-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "(SO0320-dataops-crawler) Modern Data Architecture Accelerator. Version 1.0.0",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "/test-org/test-domain/dataops-project-test/crawler/name/test-crawler",
          "Tags": {
            "mdaa_cdk_app": "dataops-crawler",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-glue-crawler-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Workflow Crawler failure events",
          "EventPattern": {
            "detail": {
              "crawlerName": [
                "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
              ],
              "state": [
                "Failed",
              ],
            },
            "source": [
              "aws.glue",
            ],
          },
          "Name": "test-org-test-env-test-domain-test-glue-crawler-main-t--4ff29561",
          "State": "ENABLED",
          "Targets": [
            {
              "Arn": {
                "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectTopicArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter",
              },
              "Id": "Target0",
            },
          ],
        },
        "Type": "AWS::Events::Rule",
      },
      "testcrawlercrawler": {
        "Metadata": {
          "cdk_nag": {
            "rules_to_suppress": [
              {
                "id": "AwsSolutions-GL1",
                "reason": "[MDAA:@aws-mdaa/glue-constructs/lib/crawler.ts:LINE:COL] Log encryption configured via SecurityConfiguration",
              },
            ],
          },
        },
        "Properties": {
          "Classifiers": [
            {
              "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestclassifiersclassifierCsvC96584B6F00A464EAD1953AFF4B05118Parameter",
            },
          ],
          "Configuration": "{"Version":1,"CrawlerOutput":{"Partitions":{"AddOrUpdateBehavior":"InheritFromTable"}}}",
          "CrawlerSecurityConfiguration": {
            "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestsecurityConfigurationdefaultC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
          "DatabaseName": {
            "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestdatabaseNameexampledatabaseC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
          "Description": "Example for a Crawler",
          "Name": "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
          "RecrawlPolicy": {
            "RecrawlBehavior": "CRAWL_NEW_FOLDERS_ONLY",
          },
          "Role": "some-arn",
          "Tags": {
            "mdaa_cdk_app": "dataops-crawler",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-glue-crawler-main",
            "mdaa_org": "test-org",
          },
          "Targets": {
            "S3Targets": [
              {
                "Path": "s3://some-s3-bucket/path/to/crawler/target",
              },
            ],
          },
        },
        "Type": "AWS::Glue::Crawler",
      },
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5",
                    ],
                    {
                      "Ref": "BootstrapVersion",
                    },
                  ],
                },
              ],
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          },
        ],
      },
    },
  },
}
`;

exports[`dataops-crawler Snapshot Tests Glue Crawler Stack Snapshot Test 1`] = `
{
  "Description": "(SO0320-dataops-crawler) Modern Data Architecture Accelerator. Version 1.0.0",
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestclassifiersclassifierCsvC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/classifiers/classifierCsv",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestdatabaseNameexampledatabaseC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/databaseName/example-database",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestdeploymentRoledefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/deploymentRole/default",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestkmsArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/kmsArn/default",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectBucketdefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/projectBucket/default",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectTopicArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/projectTopicArn/default",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaindataopsprojecttestsecurityConfigurationdefaultC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/dataops-project-test/securityConfiguration/default",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Stack description parameter to update on version changes",
        "Name": "/test-org/test-domain/test-glue-crawler-main/aws-solution",
        "Tags": {
          "mdaa_cdk_app": "dataops-crawler",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-glue-crawler-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": "(SO0320-dataops-crawler) Modern Data Architecture Accelerator. Version 1.0.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "/test-org/test-domain/dataops-project-test/crawler/name/test-crawler",
        "Tags": {
          "mdaa_cdk_app": "dataops-crawler",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-glue-crawler-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Workflow Crawler failure events",
        "EventPattern": {
          "detail": {
            "crawlerName": [
              "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
            ],
            "state": [
              "Failed",
            ],
          },
          "source": [
            "aws.glue",
          ],
        },
        "Name": "test-org-test-env-test-domain-test-glue-crawler-main-t--4ff29561",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestprojectTopicArndefaultC96584B6F00A464EAD1953AFF4B05118Parameter",
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "testcrawlercrawler": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-GL1",
              "reason": "[MDAA:@aws-mdaa/glue-constructs/lib/crawler.ts:LINE:COL] Log encryption configured via SecurityConfiguration",
            },
          ],
        },
      },
      "Properties": {
        "Classifiers": [
          {
            "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestclassifiersclassifierCsvC96584B6F00A464EAD1953AFF4B05118Parameter",
          },
        ],
        "Configuration": "{"Version":1,"CrawlerOutput":{"Partitions":{"AddOrUpdateBehavior":"InheritFromTable"}}}",
        "CrawlerSecurityConfiguration": {
          "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestsecurityConfigurationdefaultC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "DatabaseName": {
          "Ref": "SsmParameterValuetestorgtestdomaindataopsprojecttestdatabaseNameexampledatabaseC96584B6F00A464EAD1953AFF4B05118Parameter",
        },
        "Description": "Example for a Crawler",
        "Name": "test-org-test-env-test-domain-test-glue-crawler-main-test-crawler",
        "RecrawlPolicy": {
          "RecrawlBehavior": "CRAWL_NEW_FOLDERS_ONLY",
        },
        "Role": "some-arn",
        "Tags": {
          "mdaa_cdk_app": "dataops-crawler",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-glue-crawler-main",
          "mdaa_org": "test-org",
        },
        "Targets": {
          "S3Targets": [
            {
              "Path": "s3://some-s3-bucket/path/to/crawler/target",
            },
          ],
        },
      },
      "Type": "AWS::Glue::Crawler",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
